commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/TestUtils.java

```diff
@@ -410,4 +412,38 @@ public static Complex assertSame(Complex c,
         Assertions.assertEquals(z.imag(), z2.getImaginary(), () -> "Unary operator mismatch: " + name + " imaginary");
         return z;
     }
+
+    /**
+     * Assert the operation on the complex number is <em>exactly</em> equal to the operation on
+     * complex real and imaginary parts.
+     *
+     * @param c Input complex number.
+     * @param name Operation name.
+     * @param operation1 Operation on the Complex object.
+     * @param operation2 Operation on the complex real and imaginary parts.
+     * @return Result double from the given operation.
+     */
+    public static double assertSame(Complex c,
+                                    String name,
+                                    ToDoubleFunction<Complex> operation1,
+                                    DoubleBinaryOperator operation2) {
+        final double z = operation1.applyAsDouble(c);
```

Comment:
This is not a complex. The notation `z` is typically used for complex numbers. I would rename to `r1` and `r2` (r for real, or result)